### ISSUE TITLE
[DESIGN] - Water Thermoregulation

### ISSUE THREAD

**Issue by nchareunsouk** [OpenFormTech/PeaPod#73]
**BLOCKED BY** #122 

**Completion is tracked in the linked issue.**

## Design Scope

Design an in-line dimmable thermoelectric heating/cooling mechanism for the aeroponics system.

### Purpose

The purpose of the water thermoregulation system is to heat and/or cool the water as it passes through to match a desired mist spray temperature.

### Function

Inputs:
- Water (uncontrolled temperature)
- Thermoregulation power and signal (PWM to H-bridge polarity+dim switch to Peltier)

Outputs:
- Water (controlled temperature)

It interfaces with/depends on the following systems:
- **Aeroponics** (super-system)
- Energy systems
- Automation

System operation assumes:
- Aeroponics super-system delivers water to and accepts water from the system
- All components receive power

### Method

Setup:
1. Connect inputs, outputs

Process:
1. Water temperature is measured **after** the heat pump (for feedback in the forward flow direction), sent to automation
2. Control module controls heat pump speed and direction (heating or cooling the water block)
3. Water is circulated through the water block and changes temperature accordingly

Shutdown:
1. Drain system
2. Disconnect inputs, outputs

## Features
> Each feature should be added to relevant BoMs and CADs where applicable. Design decisions should be well-supported using relevant research and calculations, and should be documented in the Solution Overview. If a feature is complex enough, create a new issue for it using this template, and tag it here.

- [ ] Heater/Cooler Unit w/ Water Block
   - [x] Peltier Modules - Determine power output (single and quantity) from max target temp, max flow rate, heat capacity of water
   - [ ] Heat Sink - Determine design from heat dissipation rate
   - [ ] Fans for external heat distribution
   - [x] Water Block
- [ ] Control system
   - [ ] #120 
   - [ ] Water temperature sensor
   - [ ] PID Control loop

## Requirements and Validation

_What does this design need to accomplish? How do we know it has accomplished this?_

List any applicable metrics with criteria/constraints, applicable to purpose, function, method, and each feature, as well as how to test them.

_TBD_

## Verification
> Each test should be performed and documented, and the testing documentation attached to the PR. The testing suite should address the full scope of the **function** (i.e. ensure inputs are met and outputs work), **method** (all steps execute correctly), and **features** (each feature performs as expected) with respect to the purpose and requirements.

_How does this design achieve its method effectively, safely, and reliably? How do we test this?_

Specific testing procedure:
_TBD_

**Additional context**

Add any other context about the design scope or additional deliverables here.

_TBD_

**Comment by JLefebvre55** [comment]
We've got something already, but let's standardize.
